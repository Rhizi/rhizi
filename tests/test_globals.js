var base = require('./base');
var util = require('../scripts/util');

var count;

function object_length(obj) {
    var count = 0;
    for (var k in obj) {
        count += 1;
    }
    return count;
}

var win_before;
var win_after;

base.run_tests({
    created: function(errors, window) {
        count = object_length(window);
        console.log('before fields count: ' + count);
        win_before = util.set_from_object(window);
    },
    done:function (errors, window) {
        var new_count = object_length(window);
        console.log('after fields count: ' + new_count);
        console.log('new fields count: ' + (new_count - count));
        win_after = util.set_from_object(window);
        console.log(util.set_diff(win_after, win_before).a_b.join(','));
    }
});
