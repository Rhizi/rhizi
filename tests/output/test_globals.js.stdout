before fields count: 187
addScript: loaded ../src/external/require.js
app: running under node
addScript: loaded ../src/app.js
test harness: starting script loading with requirejs
test harness: main loaded
Rhizi main started
rhizi: init drag-n-drop
after fields count: 195
new fields count: 8
requirejs,require,define,rhizi_require_config,$,jQuery,d3,saveAs
